test/types/string/parIter/zip.chpl

```diff
@@ -0,0 +1,39 @@
+use CTypes;
+
+enum iterTypes { default, dotItems, dotBytes };
+
+config type dataType = string;
+config param iterType = iterTypes.default;
+
+const charRange = 33..126;
+
+var buf = c_calloc(uint(8), charRange.size);
+for i in charRange do buf[i-charRange.low] = i:uint(8);
+
+var str;
+if dataType == string then
+  str = createStringWithOwnedBuffer(buf, charRange.size, charRange.size);
+else
+  str = createBytesWithOwnedBuffer(buf, charRange.size, charRange.size);
+
+var rng = charRange;
+var dom = {rng};
+
+test((getIter(str), getIter(str)),               1);
+test((getIter(str), rng),                        2);
+test((getIter(str), dom),                        3);
+test((rng,          getIter(str), dom),          4);
+test((rng,          dom,          getIter(str)), 5);
+
+proc test(zipTuple, id) {
+  forall x in zip((...zipTuple)) do writeln(id, ": ", x);
+}
+
+proc getIter(s) {
+  select iterType {
+    when iterTypes.default do return s;
+    when iterTypes.dotItems do return s.items();
+    when iterTypes.dotBytes do return s.bytes();
```

Comment:
Sorry that I forgot that I was on the hook for this PR...

Are you confident that these successfully return an unevaluated iterator that's then evaluated by `test()`'s zip(), or do they manifest an array, return that, and we iterate over the array?  (or even return the iterator but then expand it in creating the tuple?)

Unrelated to this PR (and I'm probably repeating myself), but `.items()` seems like a very nonintuitive name for this iterator to me...  :(

Reply:
After reading the code, I came back to make sure we had tests that would check both the isASCII and isUTF8 code paths.  Am I correct that this one tests the ascii case only and that the other standalone tests are the ones that test the UTF8 code paths?